core/src/main/java/org/apache/calcite/config/Lex.java

```diff
@@ -63,6 +63,13 @@ public enum Lex {
   MYSQL_ANSI(Quoting.DOUBLE_QUOTE, Casing.UNCHANGED, Casing.UNCHANGED, false,
       CharLiteralStyle.STANDARD),
 
+  /** Lexical policy similar to PostgreSQL. Quoted identifiers are preserved;
+   * Unquoted identifiers are converted to lower-case; after which, identifiers
+   * are matched case-sensitively.
+   */
+  POSTGRESQL(Quoting.DOUBLE_QUOTE, Casing.TO_LOWER, Casing.UNCHANGED, false,
```

Comment:
Where do we use this? How is it tested?

Reply:
I added a test cases for this in `LexCaseSensitiveTest` and in `LexEscapeTest`.

My integration tests are located here: https://github.com/bchapuis/calcite-postgis-tests